src/components/WebLogView/ContentPreview.tsx

```diff
@@ -0,0 +1,99 @@
+import { Flex, SegmentedControl, Box, ScrollArea } from '@radix-ui/themes'
+import { useState, useEffect } from 'react'
+
+import { ReadOnlyEditor } from '../Monaco/ReadOnlyEditor'
+
+import { useGoToContentMatch } from './Details.hooks'
+import { Preview } from './ResponseDetails/Preview'
+import { Raw } from './ResponseDetails/Raw'
+
+type ContentPreviewProps = {
+  content: string
+  contentType: string
+  format: string
+  rawContent: string | undefined
+}
+
+export function ContentPreview({
+  format,
+  content,
+  rawContent,
+  contentType,
+}: ContentPreviewProps) {
+  const [selectedTab, setSelectedTab] = useState('content')
+  const { searchString, index, reset } = useGoToContentMatch()
+
+  useEffect(() => {
+    setSelectedTab('content')
+  }, [format])
+
+  // Reset search string on unmount
+  useEffect(() => {
+    return reset
+  }, [reset])
+
+  if (isMedia(format)) {
+    return (
+      <Box height="100%">
+        <Preview format={format} content={content} contentType={contentType} />
+      </Box>
+    )
+  }
```

Comment:
There could be POST requests with content type `image/jpeg` or similar mime types, so I kept this here.